elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -389,12 +396,22 @@ public boolean updateRelation(String fieldName, Set<PersistentResource> resource
                 getRelationUncheckedUnfiltered(fieldName));
         boolean isUpdated;
         if (type.isToMany()) {
-            checkFieldAwareDeferPermissions(
-                    UpdatePermission.class,
-                    fieldName,
-                    resourceIdentifiers.stream().map(PersistentResource::getObject).collect(Collectors.toList()),
-                    resources.stream().map(PersistentResource::getObject).collect(Collectors.toList())
-            );
+            if (isNewlyCreated()) {
+                checkFieldAwareDeferPermissions(
+                        CreatePermission.class,
+                        fieldName,
+                        resourceIdentifiers.stream().map(PersistentResource::getObject).collect(Collectors.toList()),
```

Comment:
Can we consolidate this input and the one following?

Reply:
Can we refactor the following segment even more into it's own function? Seen a lot of occurences in persistence resource:
```
if(isNewlyCreated()) {
    checkFieldAwareDeferPermissions( CreatePermission.class, ...);
} else {
    checkFieldAwareDeferPermissions( UpdatePermission.class, ...);
}
```

Reply:
Maybe make a function for this.

Reply:
This definitely wants to be a function.